engine/src/main/java/org/terasology/rendering/nui/editor/NUIEditorScreen.java

```diff
@@ -0,0 +1,155 @@
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.asset.UIElement;
+import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.widgets.UIDropdownScrollable;
+import org.terasology.rendering.nui.widgets.UITreeView;
+import org.terasology.rendering.nui.widgets.models.JsonTree;
+import org.terasology.rendering.nui.widgets.models.JsonTreeConverter;
+import org.terasology.rendering.nui.widgets.models.JsonTreeNode;
+import org.terasology.rendering.nui.widgets.models.Tree;
+
+import java.awt.*;
+import java.awt.datatransfer.*;
+import java.io.File;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class NUIEditorScreen extends CoreScreenLayer {
+    private Logger logger = LoggerFactory.getLogger(NUIEditorScreen.class);
+
+    @In
+    private NUIManager nuiManager;
+
+    @In
+    private AssetManager assetManager;
+
+    private List<ResourceUrn> uiUrns = Lists.newArrayList();
+
+    private UIDropdownScrollable assetList;
+    private ResourceUrn selectedUrn;
+    private UITreeView treeView;
+
+    @Override
+    public void initialise() {
+        uiUrns.addAll(assetManager
+                .getAvailableAssets(UIElement.class)
+                .stream()
+                .collect(Collectors.toList()));
+        uiUrns.sort(Comparator.comparing(ResourceUrn::toString));
+
+        assetList = find("assetList", UIDropdownScrollable.class);
+        assetList.setOptions(uiUrns);
+        assetList.bindSelection(new Binding<ResourceUrn>() {
+            @Override
+            public ResourceUrn get() {
+                return selectedUrn;
+            }
+
+            @Override
+            public void set(ResourceUrn value) {
+                if (selectedUrn != value) {
+                    updateTreeView(value);
+                }
+                // TODO: load the screen being edited; display the editor above it
+                selectedUrn = value;
+            }
+        });
+
+        treeView = find("treeView", UITreeView.class);
+
+        WidgetUtil.trySubscribe(this, "copyButton", button -> {
+            if (selectedUrn != null) {
+                JsonElement json = JsonTreeConverter.deserialize(treeView.getModel().getItem(0).getRoot());
+                String jsonString = new GsonBuilder().setPrettyPrinting().create().toJson(json);
+
+                Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
```

Comment:
I think a comment should be added that explains why you didn't use the ClipboardManager here